mlflow/models/evaluation/base.py

```diff
@@ -775,81 +775,157 @@ def _start_run_or_reuse_active_run():
-                "`evaluator` argument. If your evaluator requires additional configuration, "
-                "ensure that it is provided by specifying the `evaluator_config` argument."
-            )
-        if evaluator_config is not None:
-            conf_dict_value_error = MlflowException(
-                message="If `evaluators` argument is None, all available evaluators will be used. "
-                "If only the default evaluator is available, the `evaluator_config` argument is "
-                "interpreted as the config dictionary for the default evaluator. Otherwise, the "
-                "`evaluator_config` argument must be a dictionary mapping each evaluator's name "
-                "to its own evaluator config dictionary.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
-            if evaluator_name_list == ["default"]:
-                if not isinstance(evaluator_config, dict):
-                    raise conf_dict_value_error
-                elif "default" not in evaluator_config:
-                    evaluator_name_to_conf_map = {"default": evaluator_config}
-                else:
-                    evaluator_name_to_conf_map = evaluator_config
-            else:
-                if not check_nesting_config_dict(evaluator_name_list, evaluator_config):
-                    raise conf_dict_value_error
-                evaluator_name_to_conf_map = evaluator_config
-        else:
-            evaluator_name_to_conf_map = {}
+        # If no evaluators are specified, use all available evaluators.
+        evaluators = list(rg._registry.keys())
+
+        evaluator_config = evaluator_config or {}
+        if evaluator_config is not None and not any(
+            name in evaluator_config for name in evaluators
+        ):
+            # If evaluator config is passed but any of available evaluator key is not
+            # in the evaluator config, we assume the evaluator config to be a flat dict,
+            # which is globally applied to all evaluators.
+            evaluator_config = {ev: evaluator_config for ev in evaluators}
+
+        # Filter out evaluators that cannot evaluate the model type.
+        resolved = []
+        for name in evaluators:
+            evaluator = rg.get_evaluator(name)
+            config = evaluator_config.get(name, {})
+            if evaluator.can_evaluate(model_type=model_type, evaluator_config=config):
+                resolved.append(EvaluatorBundle(name=name, evaluator=evaluator, config=config))
+
+        # If any of built-in evaluator can apply, skip "default" evaluator.
+        default = next((ev for ev in resolved if ev.name == "default"), None)
+        non_default_builtins = [
+            ev for ev in resolved if ev.name != "default" and rg.is_builtin(ev.name)
+        ]
+        if default and non_default_builtins:
+            resolved.remove(default)
+            # Apply default config to non-default built-in evaluators if they don't have config.
```

Comment:
Is `default` equivalent to `built-in`?

Reply:
No it was previously equal to built-in, but now we have other built-in evaluators like `ClassifierEvaluator` in addition to `DefaultEvaluator`. The tricky thing here is that user will still use `"default"` for the previous meaning, so we have to replicate that config to the new built-in evaluators.

Reply:
Got it. What are non-default built-in evaluators? Are they use-defined built-in evaluators?

Reply:
They are built-in evaluators except `DefaultEvaluator`. So basically `ClassifierEvaluator` + `RegressorEvaluator` + `ShapEvaluator`.

Reply:
```suggestion
            # Apply default config to non-default built-in evaluators (e.g., ClassifierEvaluator) if they don't have config.
```

can we update the comment for clarity?

Reply:
Sure!